Feature/user creation

eugenechia95 opened the pull request:
### Context
- We need to create a different view for existing admin users to invite users
- Related PR: https://github.com/sutd-rms/burger/pull/28

### Changes
- Removed Invitation Model and all relationships with it
- Created migration files for it
- Created endpoint and view for inviting user. Feel free to test it by using an authorization token for an admin user and calling the endpoint with an email field to your own email address
- **Link in email wont work because frontend hasnt been deployed to production yet**

![image](https://user-images.githubusercontent.com/39354695/85939302-bee8d280-b946-11ea-9d10-35988b525bdf.png)

### Considerations
- Need to look into the CORS headers in the future (which is likely never). For now, it is set to allow all, which will lead to security issues in the future.

gorold commented on secretsauce/apps/account/models.py:
```diff
@@ -69,15 +55,12 @@ class User(AbstractUser):
-    invitation = models.OneToOneField(
-        Invitation,
-        on_delete=models.CASCADE,
+        null=True
```
Should a User be able to not belong to a company?

eugenechia95 replied:
Wanted to make it more flexible but on hindsight, I think it's best to set some rules for Users. Will change it back to have to include company.

gorold commented on secretsauce/apps/account/views.py:
```diff
@@ -10,62 +11,43 @@
 
-    queryset = Invitation.objects.all()
-    serializer_class = InvitationSerializer
+    def create(self, request):
+        password = generatePassword()
+        user_email = request.data['email']
+        user = User.objects.create_user(
+            user_email,
+            password
+        )
+        user.save()
+        serializer=UserCreateSerializer(user)
```
The creation of users is abit insecure, because there are no checks being done on the data being sent. For example, I can call "email=qweqwe", which is not a valid email, but an account is created. 

Also, I am able to send in the email of an existing account, which overwrites the existing account.

eugenechia95 replied:
Hmmm I thought django's EmailField would help to validate that the email provided is a valid email and coz it's supposed to be unique, it wont allow for overriding. Let me look into this.

gorold replied:
I think call User.objects.create(...) instead of create_user(...)

gorold replied:
Actually I'm not too sure about that ^. But maybe can try adding the serializer_class to do the field checks

eugenechia95 replied:
https://stackoverflow.com/questions/6039443/primary-key-and-unique-key-in-django
Can create a PR for this feature on django page LOL

gorold commented:
Apart from those above, I think its good to go!

gorold commented on secretsauce/apps/account/urls.py:
```diff
@@ -16,8 +16,7 @@
     path('company/<uuid:pk>', views.CompanyDetail.as_view()),
-]
\ No newline at end of file
+    path('inviteuser', views.InviteUser.as_view(), name='user_creation')
```
Maybe change to 'inviteuser/' to standardize?

eugenechia95 commented:
- Changed User Model to require company again
- Changed User Model to have uuid as primary key. Previous bug where I can overwrite existing account is fixed
- Strange bug exists where EmailField doesnt do a validation check when we create User Model instance via the views. Had to manually call `validate_email` method to check if email is valid. I suspect it has to do with django's orm mapping to sqlite.
